pkg/reconciler/revision/resources/deploy.go

```diff
@@ -210,6 +226,13 @@ func makePodSpec(rev *v1.Revision, cfg *config.Config) (*corev1.PodSpec, error)
 		}
 	}
 
+	if cfg.Deployment.EnablePodAntiAffinityRule && cfg.Features.PodSpecAffinity == apiconfig.Disabled {
+		revisionLabelValue := rev.Labels[serving.RevisionLabelKey]
```

Comment:
fixing this condition, since the revision itself doesn't have the label "serving.knative.dev/revision"